src/vs/code/electron-main/app.ts

```diff
@@ -359,6 +359,11 @@ export class CodeApplication extends Disposable {
 		// We handle uncaught exceptions here to prevent electron from opening a dialog to the user
 		setUnexpectedErrorHandler(error => this.onUnexpectedError(error));
 		process.on('uncaughtException', error => {
+			// Don't process the exception if exit is happening as it can cause the main process to
+			// hang while awaiting an IPC reply.
```

Comment:
I would suggest to rather listen to `onWillShutdown` as a way to know for sure that we are going down. The `quitRequested` might be `true` even though a window still vetos the shutdown.

Reply:
test